prow/spyglass/viewers/junit/junit_template.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package junit
+
+import (
+	"html/template"
+)
+
+var junitTemplateText = `{{$numF := len .Failed}}
```

Comment:
I don't think we have talked about this yet in earnest ... I really think iterating on this code will be a nightmare. It's fairly complex and the way we have it checked in, no editor will be using appropriate syntax highlighting for it, etc -- can we send HTML + CSS + JS and a data payload so at least the generation of the HTML is in JS and self-sufficient?

Reply:
Agreed, the Javascript problem is brought up in #8967. In terms of iterating, I am very much open to better suggestions, but as I've said before, I wont be addressing viewer-specific comments on this PR. I think each of the viewers is important enough to warrant their own PRs (and this PR does not profess Spyglass as the canonical way to view jobs in Prow), so I'll leave this and other comments on viewers for later.

Reply:
I don't think agreeing that a large problem exists but deciding to check in the code and punt on solving it is a reasonable approach. What if you decide you don't want to work on this anymore once this PR merges but before any future ones do? Let's not check in code that we know will require significant thought and potential re-work in thefuture.

Reply:
Ok, let's see what we can do. The main problem that I'm encountering here is that If I put the HTML/CSS/JS in separate files their paths need to be relative to the `deck` deployment (likely the static files location), which ruins the self-contained nature of the viewers. I'm not aware of a pattern that allows me to get around this.

Additionally, reviewing code from the standpoint of "what if you dont want to work on this anymore" is not helpful.

Reply:
Is a lack of syntax highlighting the only problem you are describing here or is there more to it than that? If the problem is just that editors won't highlight syntax properly that doesn't seem like a large or blocking problem.
If there is some deeper problem that makes these viewers untenable then I agree that we should resolve it here or omit the viewers from the PR until they are suitable for merge.

Reply:
FWIW many editors will syntax highlight mixed source code.. but go templates are only valid syntax as go templates. They are not guaranteed to be valid HTML prior to execution, nor should they. It is also quite useful to pre-render to HTML instead of sending data + static html + js for many use cases.

I'm happy to edit this code in the future, and I do think it is preferable to keep viewers self contained. I don't think this will be troublesome to rework, this is pretty standard. EG `hugo` which is popular (and used by our docs!) is basically a ton of stuff on top of go templates, and all docs files are essentially go template-enabled markdown.

Reply:
> Additionally, reviewing code from the standpoint of "what if you dont want to work on this anymore" is not helpful.

I think it kind of is. It is important to leave the master branch in a clean, working state. We don't want to commit half implemented or unmaintainable code so it is usually best to merge only the components that are complete and omit components that aren't complete until they are sufficiently polished (barring an emergency).
If the only problem here is syntax highlighting though, these viewers seem sufficiently complete to merge to me.

Reply:
Every single time I have worked with a system where one language was embedded in another language or configuration format, it has been a nightmare. Developing new views and iterating on them would be frustrating and error-prone. It would be simple to check something in that looks reasonable on review but is subtly broken. All of my experience just points to this being an approach that is fraught with error and if we can design something that doesn't lead us down that path I think we will be much better off.

Reply:
> It would be simple to check something in that looks reasonable on review but is subtly broken. 

This is true of all HTML. Validating HTML is non-trivial. Add in HTML + javascript rewriting it at runtime + some data format and it's definitely going to be subtly broken, as evidenced by every existing page in prow :-) Embedding a little data with a go template on the other hand is pretty easy understood compared to JS manipulating HTML, and not really any worse off on the HTML front.

New viewers do not need to use these templates, ideally we will quickly add external viewer tooling and viewer binaries can do whatever the maintainers feel are appropriate.

Reply:
I'm open to doing the leg work in implementing whatever approach we feel is best, but to me it seems that a go template is the best way to go here, at least until we get external viewers. @stevekuznetsov thoughts?

Reply:
If I'm the lone outsider on the issue, I'm ok with the Go template approach. If we find that it's frustrating to iterate on it, we can revisit in the future. I just wanted to make sure we considered the issue at hand explicitly before merging this PR.

Reply:
I'd say let's not worry about this for now (until someone complains?), or feel free to open a tracking issue